src/open_dread_rando/dread_patcher.py

```diff
@@ -30,17 +31,27 @@
 from open_dread_rando.specific_patches.static_fixes import apply_static_fixes
 from open_dread_rando.validator_with_default import DefaultValidatingDraft7Validator
 
+if typing.TYPE_CHECKING:
+    from open_dread_rando.configuration import (
+        Configuration,
+        ConfigurationCosmeticPatchesShieldVersions,
+        ConfigurationDoorPatchesItem,
+        ConfigurationNewSpawnPointsItem,
+        ConfigurationPickupsItem,
+        ConfigurationStartingItems,
+    )
+
 T = typing.TypeVar("T")
 
 
-def _read_schema():
+def _read_schema() -> dict:
     with files_path().joinpath("schema.json").open() as f:
         return json.load(f)
 
 
-def create_custom_init(editor: PatcherEditor, configuration: dict) -> str:
+def create_custom_init(editor: PatcherEditor, configuration: Configuration) -> str:
     cosmetic_options: dict = configuration["cosmetic_patches"]["lua"]["custom_init"]
-    inventory: dict[str, int] = configuration["starting_items"]
+    inventory: ConfigurationStartingItems = configuration["starting_items"]
     starting_location: dict = configuration["starting_location"]
     starting_text: list[list[str]] = configuration.get("starting_text", [])
     configuration_identifier: str = configuration["configuration_identifier"]
```

Comment:
Sometimes I just removed because the type can be inferred automatically from the argument. Sometimes I just fixed it. I wasn't exactly consistent. 

It's the classic thing of poluting the codebase with typing that can be inferred to make it easier when reading code in a bad ui